hnn_core/cell_response.py

```diff
@@ -476,3 +505,37 @@ def read_spikes(fname, gid_ranges=None):
         cell_response.update_types(gid_ranges)
 
     return cell_response
+
+
+def read_spikes_hdf5(fname, gid_ranges=None):
+    """Read spiking activity from a hdf5 file.
+
+    Parameters
+    ----------
+    fname : str
+        Path to the hdf5 file (e.g., '<pathname>/spk.hdf5') to be read.
+    gid_ranges : dict of lists or range objects | None
+        Dictionary with keys 'evprox1', 'evdist1' etc.
+        containing the range of Cell or input IDs of different
+        cell or input types. If None, data object should contain
+        spike type.
+
+    Returns
+    ----------
+    cell_response : CellResponse
+        An instance of the CellResponse object.
+    """
+
+    # A check to ensure file exists
+    if not os.path.exists(fname):
+        raise NameError("File not found at path %s" % (fname,))
```

Comment:
https://docs.python.org/3/library/exceptions.html#FileNotFoundError ?

You should test that this exception is raised in the tests

Reply:
Sure will add this test